test/functional/mempool_updatefromblock.py

```diff
@@ -38,86 +35,52 @@ def transaction_graph_test(self, size, n_tx_to_mine=None, start_input_txid='', e
-        for i in range(0, size):
-            self.log.debug('Preparing transaction #{}...'.format(i))
-            # Prepare inputs.
-            if i == 0:
-                inputs = [{'txid': start_input_txid, 'vout': 0}]
-                inputs_value = self.nodes[0].gettxout(start_input_txid, 0)['value']
-            else:
-                inputs = []
-                inputs_value = 0
-                for j, tx in enumerate(tx_id[0:i]):
-                    # Transaction tx[K] is a child of each of previous transactions tx[0]..tx[K-1] at their output K-1.
-                    vout = i - j - 1
-                    inputs.append({'txid': tx_id[j], 'vout': vout})
-                    inputs_value += self.nodes[0].gettxout(tx, vout)['value']
-
-            self.log.debug('inputs={}'.format(inputs))
-            self.log.debug('inputs_value={}'.format(inputs_value))
-
-            # Prepare outputs.
-            tx_count = i + 1
-            if tx_count < size:
-                # Transaction tx[K] is an ancestor of each of subsequent transactions tx[K+1]..tx[N-1].
-                n_outputs = size - tx_count
-                output_value = ((inputs_value - fee) / Decimal(n_outputs)).quantize(Decimal('0.00000001'))
-                outputs = {}
-                for _ in range(n_outputs):
-                    outputs[self.nodes[0].getnewaddress()] = output_value
-            else:
-                output_value = (inputs_value - fee).quantize(Decimal('0.00000001'))
-                outputs = {end_address: output_value}
-
-            self.log.debug('output_value={}'.format(output_value))
-            self.log.debug('outputs={}'.format(outputs))
-
-            # Create a new transaction.
-            unsigned_raw_tx = self.nodes[0].createrawtransaction(inputs, outputs)
-            signed_raw_tx = self.nodes[0].signrawtransactionwithwallet(unsigned_raw_tx)
-            tx_id.append(self.nodes[0].sendrawtransaction(signed_raw_tx['hex']))
-            tx_size.append(self.nodes[0].getrawmempool(True)[tx_id[-1]]['vsize'])
-
-            if tx_count in n_tx_to_mine:
-                # The created transactions are mined into blocks by batches.
-                self.log.info('The batch of {} transactions has been accepted into the mempool.'.format(len(self.nodes[0].getrawmempool())))
-                block_hash = self.nodes[0].generate(1)[0]
-                if not first_block_hash:
-                    first_block_hash = block_hash
-                assert_equal(len(self.nodes[0].getrawmempool()), 0)
+        self.wallet.generate(1)
+        node.generate(size)
+        txs = []
+        block_ids = []
+        tx_size = []
+        for i in range(size):
+            txs.append(self.wallet.send_self_transfer(from_node=node))
+            tx_size.append(96)
+            if (i+1) % n_tx_to_mine == 0:
+                self.log.info('The batch of {} transactions has been accepted into the mempool.'
```

Comment:
Would it be useful to have this in the line above?
```suggestion                
                assert_equal(len(node.getrawmempool()), n_tx_to_mine)
```

Reply:
Shoots, I thought I had that but it was just assert for mempool after mining a block. Updated ac8b08b